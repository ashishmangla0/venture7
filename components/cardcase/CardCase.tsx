

import Image from 'next/image'
import cardCaseStyle from './cardcase.module.scss'

import { CardCaseProps } from '@/types'
import Link from 'next/link'

const CardCase = (props: CardCaseProps) => {
    const { cardWrapperClass, title, description,imageSrc,readMoreLink,linkText } = props
    return (
        <article className={`${cardCaseStyle.card} ${cardWrapperClass}`}>
            <div className={`${cardCaseStyle.card__img}`}>
                <Image
                    alt={title || ''}
                    src={imageSrc || ''}
                    layout='fill'
                    objectFit='cover'
                />
            </div>
            <div className={`${cardCaseStyle.card__content}`}>
                {title && <h3>
                    {title}
                </h3>}
                {description && <p>
                    {description}
                </p>}
                {readMoreLink && <Link className='button' href={readMoreLink} >{linkText}</Link> }
            </div>
        </article>
    )
}
CardCase.defaultProps = {
    cardWrapperClass: '',
    title: '',
    description: '',
    imageSrc:'',
    readMoreLink:'/',
    linkText: 'Read More'
}
export default CardCase