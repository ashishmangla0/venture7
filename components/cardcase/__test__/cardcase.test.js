import { render } from '@testing-library/react'
import CardCase from '../CardCase'




describe('<CardCase/>', () => {
  it('renders CardCase Snapshot', () => {
    const { container } = render(<CardCase />)
    expect(container).toMatchSnapshot();
  })
  test('renders CardCase Component with Default values', () => {
    const { getByAltText, getByText, container } = render(<CardCase />);
    const actionButton = getByText('Read More');

    expect(actionButton).toBeInTheDocument();
    //actionButton.toBeInTheDocument();
    expect(container.querySelector('h3')).toBeNull();
    expect(container.querySelector('p')).toBeNull();

    //expect(container.querySelector('h3')).not.toBeNull();


  });
});