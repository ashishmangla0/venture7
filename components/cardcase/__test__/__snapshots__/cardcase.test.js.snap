// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardCase/> renders CardCase Snapshot 1`] = `
<div>
  <article
    class="card "
  >
    <div
      class="card__img"
    >
      <img
        alt=""
        data-nimg="fill"
        decoding="async"
        src=""
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="card__content"
    >
      <a
        class="button"
        href="/"
      >
        Read More
      </a>
    </div>
  </article>
</div>
`;
